### DISCUSSION TITLE
Zaptec Go support?

### DISCUSSION THREAD

**Discussion by JiePieWie** [lbbrhzn/ocpp#835] [Q&A]
We are thinking about buying a Zaptec Go for charging our EV. According to the [documentation](https://zendesk.zaptec.com/hc/en-001/sections/4529640694417-Guides) it supports ocpp. Will this also work with this HA integration? Anybody any experience with this charger and this integration?

**Comment by oleost** [comment]
I have same question

**Comment by FSund** [comment]
I've been trying to set up Zaptec Go, but I keep getting the following error

> Ladestasjonen er ikke tilkoblet OCPP-server.
> 16. okt. 2024, 18:46
>
> Unable to connect; ws://192.168.1.202:9000/ (status: ConnectionError, errorCode: <null>, context: Unexpected WebSocketException); Exception: System.Net.WebSockets.WebSocketException The server returned status code '400' when status code '101' was expected.

Any idea what I'm doing wrong? Here's my config

![image](https://github.com/user-attachments/assets/144f798f-e2b6-4d5e-98e2-eb357086fc74)

**Reply by FSund** [reply]
Just found the following error in my logs

> Protocols mismatched | expected Subprotocols: ['ocpp1.6'], but client supports | Closing connection

I tried enabling the `Skip OCPP schema validation` option in the Integration, but it doesn't seem like it helped.

**Comment by Revilo-S** [comment]
Any news on this problem? I try to set up a Zaptec Go 2 but I get the same error.

**Comment by FSund** [comment]
Yes, sorry for not updating before. Zaptec says that the OCPP connection goes via the cloud, so the server needs to be exposed to the web. No local connection... They have something called OCPP Native, but that is reserved for larger corporate customers.

I have not tried with a publicly exposed server. I resorted to using this integration instead, since OCPP relies on the cloud anyway https://github.com/custom-components/zaptec

**Comment by Revilo-S** [comment]
Thanks a lot for your reply. This integration was my first try. I'd like to control the box locally so I gave the ocpp integration a shot.
I will contact Zaptec support again. Maybe they will consider implementing local communication in the future.

**Comment by Revilo-S** [comment]
Zaptec support responded instantly :-)
"We haven't yet integrated native OCPP at the box level on the Zaptec Go2.
This currently only runs on the Zaptec Go.
However, we are already working on bringing native OCPP to the Go2."
I hope, native OCPP will also available for us private customers.
